models/stock_lot.py

```diff
@@ -0,0 +1,46 @@
+from odoo import models, fields,api
+
+
+class StockProductionLot(models.Model):
+    _inherit = "stock.lot"
+
+    is_flower = fields.Boolean(related='product_id.is_flower', readonly=True)
+    water_ids = fields.One2many("flower.water", "serial_id")
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            product = self.env["product.product"].browse(vals["product_id"])
+            if product.sequence_id:
+                vals["name"] = product.sequence_id.next_by_id()
+        return super().create(vals_list)
```

Comment:
this needs to check if the product is a flower

Reply:
Done..